src/cpuid/src/common.rs

```diff
@@ -1,16 +1,200 @@
 // Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
+#![allow(clippy::restriction)]
 
-#[cfg(target_arch = "x86")]
-use std::arch::x86::{CpuidResult, __cpuid_count, __get_cpuid_max};
-#[cfg(target_arch = "x86_64")]
-use std::arch::x86_64::{CpuidResult, __cpuid_count, __get_cpuid_max};
+#[cfg(cpuid)]
+use crate::bit_helper::BitHelper;
+
+macro_rules! bit_range {
+    ($msb_index:expr, $lsb_index:expr) => {
+        crate::bit_helper::BitRange {
+            msb_index: $msb_index,
+            lsb_index: $lsb_index,
+        }
+    };
+}
 
-#[cfg(target_arch = "x86_64")]
-use kvm_bindings::CpuId;
+// Basic CPUID Information
+#[allow(clippy::doc_markdown, dead_code)]
```

Comment:
Why is this dead_code allowed? Maybe add a comment?

Reply:
Originally here as I developed to avoid warnings.

I've removed this and the dead code.